include/biosHandler.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+#include <sdbusplus/asio/connection.hpp>
+#include <sdbusplus/bus.hpp>
+
+namespace vpd
+{
+/**
+ * @brief A class to operate upon and back up some of the BIOS attributes.
+ *
+ * The class initiates call backs to listen to PLDM service and changes in some
+ * of the BIOS attributes.
+ * On a factory reset, BIOS attributes are initialized by PLDM. this class waits
+ * until PLDM has grabbed a bus name before attempting any syncs.
+ *
+ * It also initiates API to back up those data in VPD keywords.
+ */
+class BiosHandler
+{
+  public:
+    // deleted APIs
+    BiosHandler() = delete;
+    BiosHandler(const BiosHandler&) = delete;
+    BiosHandler& operator=(const BiosHandler&) = delete;
+    BiosHandler& operator=(BiosHandler&&) = delete;
+    ~BiosHandler() = default;
+
+    /**
+     * @brief Constructor.
+     *
+     * @param[in] connection - Asio connection object.
+     */
+    BiosHandler(std::shared_ptr<sdbusplus::asio::connection>& connection) :
```

Comment:
i_connection

Reply:
have taken care in the incremental commit on top if this PR.